File: Luogu/Greedy/P1094.cc
#include <iostream>
#include <vector>
#include <algorithm>

int main(void) {
    int w = 0, n = 0;
    std::cin >> w >> n;

    std::vector<int> souvenirs(n, 0);
    for (int i = 0; i < n; i ++) {
        std::cin >> souvenirs[i];
    }

    std::sort(souvenirs.begin(), souvenirs.end());

    int i = 0, j = n - 1;
    int groups = 0;
    while (i < j) {
        if (souvenirs[i] + souvenirs[j] <= w) {
            groups ++;
            i ++;
            j --;
        } else {
            j --;
        }
    }

    int res = n - groups;

    std::cout << res << std::endl;

    return 0;
}

File: Luogu/Greedy/P1094.py
w = 0
in_str = list(input().split(" "))
for str in in_str:
    if ('' != str):
        w = int(str)

n = 0
in_str = list(input().split(" "))
for str in in_str:
    if ('' != str):
        n = int(str)

souvenirs = []
for i in range(n):
    in_str = list(input().split(" "))
    for str in in_str:
        if ('' != str):
            souvenirs.append(int(str))

souvenirs.sort(reverse=False)

i = 0
j = n - 1
groups = 0
while i < j:
    if souvenirs[i] + souvenirs[j] <= w:
        groups += 1
        i += 1
        j -= 1
    else:
        j -= 1
       
print(n - groups)